python/ray/ml/examples/pytorch/permuted_mnist_domain_incremental_learning_example.py

```diff
@@ -0,0 +1,297 @@
+
+    for dataset in dataset_stream:
+        trainer = TorchTrainer(
+            train_loop_per_worker=train_loop_per_worker,
+            train_loop_config={
+                "num_epochs": 2,
+                "learning_rate": 0.001,
+                "momentum": 0.9,
+                "batch_size": 32,
+            },
+            scaling_config={"num_workers": 2, "use_gpu": False},
+            datasets={"train": dataset},
+            preprocessor=BatchMapper(fn=preprocess_images),
+            resume_from_checkpoint=latest_checkpoint,
+        )
+        result = trainer.fit()
+        latest_checkpoint = result.checkpoint
+
+    return latest_checkpoint
+
+
+def predict(checkpoint: Checkpoint, test_dataset_stream: Iterator[ray.data.Dataset]):
+    """Runs prediction of the incrementally trained model on stream of test datasets.
+
+    Args:
+        checkpoint: The checkpoint containing the trained model.
+        test_dataset_stream: A stream of Ray Datasets to test on.
+    """
+
+    acc_results = []
+    batch_predictor = BatchPredictor.from_checkpoint(
+        checkpoint=checkpoint,
+        predictor_cls=TorchPredictor,
+        model=SimpleMLP(num_classes=10),
+    )
+
+    for test_dataset in test_dataset_stream:
+        model_output = batch_predictor.predict(
+            data=test_dataset, feature_columns=["image"], unsqueeze=False
+        )
+        # Postprocess model outputs.
+        # Convert logits outputted from model into actual class predictions.
+        # TODO: Find a better way to do this. This is pretty ugly right now.
+        # TODO: See if Predictors can accept "postprocessing" transforms to be applied
+        #  directly after model output.
+
+        def convert_logits_to_classes(df):
+            best_class = df["predictions"].map(lambda x: np.array(x).argmax())
+            df["predictions"] = best_class
+            return df
+
+        prediction_results = model_output.map_batches(
+            convert_logits_to_classes, batch_format="pandas"
+        )
+
+        # Then, for each prediction output, see if it matches with the ground truth
+        # label.
+        # TODO: Find a better way to do this. This is pretty ugly right now.
+        zipped_dataset = test_dataset.zip(prediction_results)
+
+        def calculate_prediction_scores(df):
+            df["correct"] = df["predictions"] == df["label"]
+            return df
+
+        correct_dataset = zipped_dataset.map_batches(
+            calculate_prediction_scores, batch_format="pandas"
+        )
+
+        acc_results.append(correct_dataset.sum(on="correct") / test_dataset.count())
+
+    print(acc_results)
+    return acc_results
+
+
+if __name__ == "__main__":
+    dataset_stream = PermutedMNISTStream(n_tasks=3)
+    checkpoint = incremental_train(dataset_stream.generate_train_stream())
+    checkpoint.to_directory("./checkpoint")
+    predict(
+        Checkpoint.from_directory("./checkpoint"), dataset_stream.generate_test_stream()
```

Comment:
so I don't have the full context of this example.
but for an incremental training example, I imagine it will be more interesting if we do it like:

train(dataset_1)
test(dataset_test)
train(dataset_2)
test(dataset_test)
...

you don't just to see the progress incrementally.
in production, I also imagine that you will deploy incremental checkpoints, then train some more., then deploy again.

Reply:
Done!